### PR TITLE
feat: Add /nodes/{nodeLabel}/dependencies endpoint

### PR DESCRIPTION
**Pull request by andrewelamb**
Depends on https://github.com/Sage-Bionetworks/sage-monorepo/pull/1903
fixes [fds-782](https://sagebionetworks.jira.com/browse/FDS-782)

This PR:

- Adds the /nodes/{nodeLabel}/dependencies endpoint (formerly  /explorer/get_node_dependencies)
- Adds the node, nodesPage, and nodeLabel schema
- Adds the component and componentsPage schema (not needed for this PR, but kept for future ones)
- Adds returnDisplayNames and returnOrderedBySchema parameters

Files to Check:

- Everything in libs/schematic/api-description/src/ 
- Everything in apps/schematic/api/schematic_api/test/
- apps/schematic/api/schematic_api/controllers/schema_controller_impl.py

### CODE REVIEW

File: libs/schematic/api-description/build/openapi.yaml [hunk 1]
```diff
@@ -94,6 +94,28 @@ paths:
                 $ref: '#/components/schemas/AttributesPage'
         '500':
           $ref: '#/components/responses/InternalServerError'
+  /components/{componentLabel}/parents:
```
Comment (mialy-defelice) [inline libs/schematic/api-description/build/openapi.yaml:97]: As noted elsewhere this is not a component specific endpoint and can be used for any node in the graph, would likely only return viable results for attributes, but if an attribute was also a valid value results would be returned if the attributes would have dependencies.

File: libs/schematic/api-description/build/openapi.yaml [hunk 2]
```diff
@@ -312,6 +334,42 @@ components:
             - attributes
       x-java-class-annotations:
         - '@lombok.Builder'
+    ReturnDisplayNames:
+      description: Whether or not to return the display names of the component, otherwise the label
+      type: boolean
+      default: true
+    ReturnOrderedBySchema:
+      description: Whether or not to order the components by their order in the schema, otherwise random
```
Comment (mialy-defelice) [inline libs/schematic/api-description/build/openapi.yaml:342]: same comment as above, many of the things returned are actually attributes.

**Review by mialy-defelice — CHANGES_REQUESTED** [review]
My overall comment is about the use of the terms within the descriptors and to make sure its clear what is happening and what is expected throughout. I am happy to chat about this if you want.

**Comment by andrewelamb** [comment]
@mialy-defelice This is ready to re-check. Thanks!

File: apps/schematic/api/schematic_api/controllers/schema_controller_impl.py [hunk 3]
```diff
@@ -62,3 +64,75 @@ def list_component_attributes(
...
+
+
+@handle_exceptions
+def list_node_dependencies(
+    node_label: str,
+    schema_url: str,
+    return_display_names: bool = True,
+    return_ordered_by_schema: bool = True,
+) -> tuple[Union[NodesPage, BasicError], int]:
+    """Lists the components that the input component is dependent on
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/controllers/schema_controller_impl.py:103]: this should read as "Lists the attributes that the input attribute is dependent on"

File: apps/schematic/api/schematic_api/controllers/schema_controller_impl.py [hunk 4]
```diff
@@ -62,3 +64,75 @@ def list_component_attributes(
...
+        node_label (str): The label of the node to get dependencies for
+        schema_url (str): The URL of the schema in json form
+        return_display_names (bool): Whether or not to return the display names of the dependencies,
+          otherwise the label
+        return_ordered_by_schema (bool):Whether or not to order the dependencies by their order in
+          the schema, otherwise random
+
+    Returns:
+        tuple[Union[ComponentsPage, BasicError], int]: A tuple
+          The first item is either the components or an error object
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/controllers/schema_controller_impl.py:115]: substitute components for either attributes or nodes.

**Review by mialy-defelice — CHANGES_REQUESTED** [review]

**Review by mialy-defelice — APPROVED** [review]

**Pull request merged**